packages/custom_lint/lib/src/output/json_output_format.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:convert';
+
+import 'package:analyzer_plugin/protocol/protocol_common.dart';
+import 'package:cli_util/cli_logging.dart';
+
+import 'output_format.dart';
+
+/// The JSON output format.
+///
+/// Code is an adaption of the original Dart SDK JSON format.
+/// See: https://github.com/dart-lang/sdk/blob/main/pkg/dartdev/lib/src/commands/analyze.dart
+class JsonOutputFormat implements OutputFormat {
+  @override
+  void render({
+    required Iterable<AnalysisError> errors,
+    required Logger log,
+  }) {
+    Map<String, dynamic> location(
```

Comment:
Those utils should be extracted to top-level private functions

And don't use => but { return}

Reply:
The whole logic is pretty much a copy from the Dart SDK. I didn't want to change too much here so it is easier to adapt to changes in the SDK in the future. I only adapted minor things to please the linter.

I agree with all your change requests but left this intentionally the way it is.
If you think it is still better to adapt, I will change it.